File: Bidapp Demo App - ObjectiveC/bidappPlayground/FullscreenLoadDelegate.m
//
//  FullscreenLoadDelegate.m
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

#import "FullscreenLoadDelegate.h"
#import "tools.h"
#import <bidapp/BIDAdInfo.h>
#import <bidapp/BIDAdFormat.h>

@implementation FullscreenLoadDelegate

#pragma mark - BIDFullscreenLoadDelegate

- (void)didLoadAd:(BIDAdInfo*)adInfo
{
	NSLog(@"[%@] didLoadAd: %d. IsRewarded: %@ ", sessionIdShort(adInfo.loadSessionId), adInfo.networkId, (adInfo.format.isRewarded) ? @"YES" : @"NO");
}

- (void)didFailToLoadAd:(BIDAdInfo*)adInfo error:(NSError *)error
{
	NSLog(@"[%@] didFailToLoadAd: %d. ERROR: %@", sessionIdShort(adInfo.loadSessionId), adInfo.networkId, error.localizedDescription ? error.localizedDescription : error);
}

@end

File: Bidapp Demo App - SwiftUI/bidappPlayground/FullscreenLoadDelegate.swift
//
//  FullscreenLoadDelegate.swift
//  bidappPlayground
//
//  Copyright 2023 bidapp. All rights reserved.
//

import UIKit
import bidapp

class FullscreenLoadDelegate : NSObject, BIDFullscreenLoadDelegate {
    func didLoadAd(_ adInfo: BIDAdInfo) {
        let waterfallId = adInfo.waterfallId ?? ""
        let isRewardedString = adInfo.format.isRewarded() ? "yes" : "no";
        print("Bidapp fullscreen [\(waterfallId)] didLoadAd: \(adInfo.networkId) [\(adInfo)]. IsRewarded: \(isRewardedString)")
    }
    
    func didFail(toLoadAd adInfo: BIDAdInfo, error: Error) {
        let waterfallId = adInfo.waterfallId ?? ""
        let descr = error.localizedDescription
        print("Bidapp fullscreen [\(waterfallId))] didFailToLoadAd: \(adInfo.networkId) [\(adInfo)]. ERROR: \(descr)");
    }
}
